doc/source/whatsnew/v0.24.0.rst

```diff
@@ -731,6 +731,38 @@ Bug in :func:`DataFrame.to_dict` raises ``ValueError`` when used with
 
     df.to_dict(orient='index')
 
+.. _whatsnew_0240.api_breaking.incompatible_index_unions
+
+Incompatible Index Type Unions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When performing an :func:`Index.union` operation, between objects of incompatible dtypes, the result will be a base :class:`Index` of dtype `object`. 
+This behavior holds true for unions between :class:`Index` objects that previously would have been prohibited.
+The dtype of empty :class:`Index` objects will now be evaluated before performing the union as well, instead of simply returning the other :class:`Index` object.
```

Comment:
remove after the comma (instead .....); add the issue number here

Reply:
done